### PR TITLE
Query with large offsets tutorial

### PR DESCRIPTION
**Pull request by schleichardt**

### CODE REVIEW

File: sphere-models/src/it/java/io/sphere/sdk/products/queries/QueryAllTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+public class QueryAllTest extends QueryAllBase {
+    //yes, this is a small page size, but since this is tested,
+    //we create not million products for test execution speed reasons
+    //in production code it would higher
+    private static final int PAGE_SIZE = 5;
+
+    @Test
+    public void useIdPredicateInsteadOfOffset() throws Exception {
+        final ProductQuery seedQuery = ProductQuery.of()
```
Comment (butenkor) [inline sphere-models/src/it/java/io/sphere/sdk/products/queries/QueryAllTest.java]: ProductQuery can be used for both "products" and "product-projections" endpoint?

Comment (schleichardt) [inline sphere-models/src/it/java/io/sphere/sdk/products/queries/QueryAllTest.java]: No, but the same system should work with ProductProjectionQuery.

File: src/main/java/io/sphere/sdk/meta/QueryDocumentation.java [hunk 2]
```diff
@@ -185,6 +185,15 @@
 
  <p>So, for this example query with offset 2 and limit 4, the {@link io.sphere.sdk.queries.PagedQueryResult} will have offset 2, size 4 and total 15. But be careful, count can be smaller than limit in some cases; for example, if total is smaller than the limit (limit 500 but only 15 products). It can also be the case when total is not dividable by limit and the last elements are fetched (.e.g. |12 13 14| with offset 12 and limit 4).</p>
 
+ <h3 id=pagination-for-large-offsets>Pagination for large offsets</h3>
+
+ Using big offsets can slow down the query and has an impact on other requests since the DB is busy with that expensive query.
```
Comment (butenkor) [inline src/main/java/io/sphere/sdk/meta/QueryDocumentation.java]: Maybe you should mention sorting here too: https://github.com/sphereio/sphere-jvm-sdk/pull/559/files#diff-c61b74382a7125d79530f3ebcc8cb92eR61

File: src/main/java/io/sphere/sdk/meta/QueryDocumentation.java [hunk 3]
```diff
@@ -185,6 +185,15 @@
...
+ See <a href="http://stackoverflow.com/a/7228190">http://stackoverflow.com/a/7228190 for more information.</a>
+
+ <p>For some cases, like iterating through a whole collection for imports/exports the offset parameter can be avoided by using a predicate id &gt; $lastId.</p>
```
Comment (butenkor) [inline src/main/java/io/sphere/sdk/meta/QueryDocumentation.java]: Maybe you should mention sorting here too: https://github.com/sphereio/sphere-jvm-sdk/pull/559/files#diff-c61b74382a7125d79530f3ebcc8cb92eR61

**Comment by butenkor** [comment]
:+1: for all the tests :)

**Pull request merged**